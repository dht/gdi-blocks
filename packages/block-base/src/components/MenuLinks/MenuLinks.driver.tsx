import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MenuLinks, MenuLinksProps } from './MenuLinks';
import { BaseComponentDriver } from 'testing-base';

export class MenuLinksDriver extends BaseComponentDriver {
    private props: Partial<MenuLinksProps> = {};

    constructor() {
        super('MenuLinks');
    }

    when: any = {
        rendered: () => {
            render(<MenuLinks {...(this.props as MenuLinksProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(<MenuLinks {...(this.props as MenuLinksProps)} />);
        },
    };

    given: any = {
        props: (props: Partial<MenuLinksProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        WrapperClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
