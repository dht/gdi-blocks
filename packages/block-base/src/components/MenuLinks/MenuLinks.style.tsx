import styled from 'styled-components';

export const Wrapper = styled.div`
    flex: 1;
    margin-left: 50px;
`;

export const Ul = styled.ul`
    list-style: none;
    padding: 0;
    margin: 0;
    display: flex;
    flex-direction: row;
    align-items: center;
`;

export const Li = styled.li`
    padding: 0;
    margin: 0 20px;
`;

export const A = styled.a`
    text-transform: uppercase;
    font-variation-settings: 'wdth' 100, 'wght' 350;
    font-size: 18px;
    text-decoration: none;
    color: #334;
    white-space: nowrap;
`;
