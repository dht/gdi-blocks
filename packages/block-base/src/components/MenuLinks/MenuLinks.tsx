import React from 'react';
import { A, Li, Ul, Wrapper } from './MenuLinks.style';

export type MenuLinksProps = {};

export function MenuLinks(_props: MenuLinksProps) {
    return (
        <Wrapper className='MenuLinks-wrapper' data-testid='MenuLinks-wrapper'>
            <Ul>
                <Li>
                    <A href='#home'>Customers ❤️</A>
                </Li>
                <Li>
                    <A href='#home'>Movement</A>
                </Li>
                <Li>
                    <A href='#home'>Diary</A>
                </Li>
                <Li>
                    <A href='#home'>Contact</A>
                </Li>
            </Ul>
        </Wrapper>
    );
}

export default MenuLinks;
