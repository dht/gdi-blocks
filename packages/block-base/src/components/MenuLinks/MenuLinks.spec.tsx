import { MenuLinksDriver } from './MenuLinks.driver';
import Chance from 'chance';

const chance = new Chance();

describe('MenuLinks', () => {
    let driver: MenuLinksDriver;

    beforeAll(() => {
        driver = new MenuLinksDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const wrapperClassName = element.get.wrapperClassName();
        const innerText = element.get.label();

        expect(wrapperClassName).toContain('MenuLinks-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('MenuLinks snapshots', () => {
    let driver: MenuLinksDriver;

    beforeAll(() => {
        driver = new MenuLinksDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
