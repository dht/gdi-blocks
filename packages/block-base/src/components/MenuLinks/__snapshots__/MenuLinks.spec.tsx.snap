// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLinks snapshots should match snapshot 1`] = `
<div
  className="sc-bcXHqe ijiIm MenuLinks-wrapper"
  data-testid="MenuLinks-wrapper"
>
  <ul
    className="sc-gswNZR gplOJJ"
  >
    <li
      className="sc-dkrFOg rmVYm"
    >
      <a
        className="sc-hLBbgP fBLHEp"
        href="#home"
      >
        Customers ❤️
      </a>
    </li>
    <li
      className="sc-dkrFOg rmVYm"
    >
      <a
        className="sc-hLBbgP fBLHEp"
        href="#home"
      >
        Movement
      </a>
    </li>
    <li
      className="sc-dkrFOg rmVYm"
    >
      <a
        className="sc-hLBbgP fBLHEp"
        href="#home"
      >
        Dear diary
      </a>
    </li>
    <li
      className="sc-dkrFOg rmVYm"
    >
      <a
        className="sc-hLBbgP fBLHEp"
        href="#home"
      >
        Contact
      </a>
    </li>
  </ul>
</div>
`;
